aiokafka/record/legacy_records.py

```diff
@@ -11,6 +11,9 @@
 )
 
 
+NoneType = type(None)
```

Comment:
Where is it used?

Reply:
Lines 332 and 334. Those lines were previously broken into two:
```
if not (key is None or
    isinstance(key, (bytes, bytearray, memoryview))):
```

The use of NoneType (which for some reason isn't automatically in the global namespace) allows the expression to be condensed onto a single line:
```
if not isinstance(key, (bytes, bytearray, memoryview, NoneType)):
```

Reply:
`is None` check is the fastest opcode in Python... The isinstance(NoneType) on the other hand is slow.

Reply:
Right, but this diff only performs type checking in exceptional circumstances rather than on every `builder.append()`. Since the outcome of this block of code will always be a raised exception, I judged that the minor decrease in performance was worth the increased readability. I'm happy to switch it back if you'd prefer.